src/server.rs

```diff
@@ -100,15 +101,28 @@ impl<State: Clone + Send + Sync + 'static> Server<State> {
     /// # Ok(()) }) }
     /// ```
     pub fn with_state(state: State) -> Self {
-        let mut server = Self {
-            router: Arc::new(Router::new()),
-            middleware: Arc::new(vec![]),
-            state,
-        };
-        server.with(cookies::CookiesMiddleware::new());
-        #[cfg(feature = "logger")]
-        server.with(log::LogMiddleware::new());
-        server
+        #[cfg(any(feature = "cookies", feature = "logger"))]
```

Comment:
This seems unnecessary. Why not just add the one `#[cfg()]` without this outside wrapping? It'l compile to the same thing.

Is it just because something may warn about `mut`?

Reply:
It's because the warning about `mut` is considered an error in CI, so the pipeline fails.

Reply:
I think that @jbr will probably strongly dislike this suggestion but, an alternate and perhaps better (if to avoid warnings) way to write this is to just move the `#[cfg()]`'s into the `vec![]`:

```rust
vec![
   #[cfg(feature = "cookies")]
   cookies::CookiesMiddleware::new(),
   #[cfg(feature = "logger")]
   log::LogMiddleware::new(),
]
```

Reply:
Ah, so just instantiate the server with the given middleware rather than calling `.with`?

Reply:
Correct. Again, if we are avoiding a warning for that `mut`, then this seems to be the best way to do so.

Reply:
@belak Were you able to make this change? I'd like to ensure this gets in for tide 0.14

Reply:
Sorry about the delay - give me a few min and I'll have that in